cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java

```diff
@@ -67,8 +68,25 @@
     private boolean helpFlag;
 
     private static boolean isDirEmpty(final Path directory) throws IOException {
+
         try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(directory)) {
-            return !dirStream.iterator().hasNext();
+            //Check whether the OS is MacOS and the folder contains .DS_Store file
+            Iterator pathIterator = dirStream.iterator();
+            if (pathIterator.hasNext()) {
+
+                Path path = (Path) pathIterator.next();
+                Path filePath = path.getFileName();
+                if (filePath != null && filePath.toString().equals(ProjectDirConstants.MAC_OS_DS_STORE_FILE)) {
+
+                    return System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("mac") &&
+                            !pathIterator.hasNext();
```

Comment:
Why do we need to check this condition again? 
```java
 !pathIterator.hasNext()
```

Reply:
Otherwise, we won't be able to detect if there are more files other than the `.DS_Store`. Assume there are two files and one is the `.DS_Store`. Iterator picks the `.DS_Store` as its first item. Then there should be a way to check the additional files. In this case `!pathIterator.hasNext()` will return false, but if this is not here then the whole function will return `true` which is not the case.

Reply:
Okay got it :) So we can remove the check for the os and have only the check for the file name

Reply:
Don't we have to check the OS as well? What happens if there is a manually created `.DS_Store` file in a Linux system?

Reply:
There is a rare chance of that happening right :) Even if such a file was created we would want to generate the ballerina source file. So I think it would be better if we remove the check added for the os name.

Reply:
If that's the case will remove the OS check. Then the logic can be simplified to 

`return fileName != null && fileName.toString().equals(ProjectDirConstants.DS_STORE_FILE) && !pathIterator.hasNext();`